### DISCUSSION TITLE
About Collider

### DISCUSSION THREAD

**Discussion by sayyes304** [mlavik1/UnityVolumeRendering#131] [Q&A] [answered]
Hi Thank you for amazing project !
I have question 
Is it possible to apply collider for each 3D rendered object? 
In other words, is it possible to manage by extracting bones or organs separately?

**Comment by mlavik1** [comment, marked as answer]
Hi @sayyes304 !

So if I understand you correctly, you want to create a collider that represents a volumetric dataset? (Which format is it btw?)

My suggestion would be to generate a mesh using marching cubes, and then use it in a [MeshCollider](https://docs.unity3d.com/ScriptReference/MeshCollider.html).

You could for example use this library: https://github.com/AOT-AG/DicomToMesh
To export an .obj file. Then import that into Unity, and add it to a mesh collider.

Would that work?

If you want to have a mesh collider for only part of a dataset, I guess you'd need to extract a part of it somehow. I suppose there might be some software that can do that, but I don't know of any.
Maybe I should add support for exporting datasets? :thinking:

**Reply by sayyes304** [reply]
Thank you for response 😀 ! this is really helpful (solved this question)

**Reply by mlavik1** [reply]
Happy to hear that! :)

**Reply by sayyes304** [reply]
Can I ask more question ? sorry to bother you 😥

1)
I know the volume rendering can be largely divided into a surface-based rendering method that extracts surface models first and renders these surfaces, and a volume-based rendering method that obtains images directly from volume data without generating surface models.

If I understand correctly, in your project (I refer tutorial explaining the basic implementation), are you using both surface-based(raymarching) and volume-based rendering methods(ray tracing)?

Then, Direct Volume Rendering uses only Raytracing, and Marching Cube is only used for Isosurface Rendering?
(But the DirectVolumeRendering shader code use only ray marching... )

or, is the marching cube algorithm applied to volume raytracing, one of the direct volume rendering techniques?

2) If I understand correctly, you use marching cube for extract density of 3D voxel (not for render 3D mesh) ? and  I want to know why you saved 3D voxel data from a medical CT scan in 3D Texture instead of 3D Mesh

**Reply by mlavik1** [reply]
Hi again!

So I'm not using marching cubes for anything in this project, but instead only render using "raymarching".
The imported datasets are basically just grids of density values. Each point inside the "grid" has a density value (higher value for bone, etc.). [Marching cubes](https://en.wikipedia.org/wiki/Marching_cubes) is a method for creating polygonial meshes form this data, but as mentioned this library does not do that.
Instead the volume is imported to a 3D array of density values, and then this is written to a 3D texture so we can use it in the shader. A 3D texture is also like a 3D grid of values, so this is a very simple and natural operation - while using a mesh would require you to create that mesh first (using marching cubes or something). Also, direct volume rendering (by raymarching a 3D texture) allows you to easily use transfer functions.

There are 3 rendering modes in this project, but all of them are using raymarching. Even the "isosurface" mode, which marches along a ray until something is hit, and then draws that.

**Reply by sayyes304** [reply]
Oh I was confused about that.
Thank You ! 👍
